pkg/helm/controller.go

```diff
@@ -120,14 +131,18 @@ func (c *Controller) OnHelmChanged(key string, chart *helmv1.HelmChart) (*helmv1
 }
 
 func (c *Controller) OnHelmRemove(key string, chart *helmv1.HelmChart) (*helmv1.HelmChart, error) {
+	if chart == nil {
+		return nil, nil
+	}
 	if chart.Spec.Chart == "" {
 		return chart, nil
 	}
+
 	job, _, _ := job(chart)
 	job, err := c.jobsCache.Get(chart.Namespace, job.Name)
 
 	if errors.IsNotFound(err) {
-		_, err := c.OnHelmChanged(key, chart)
+		_, err := c.OnHelmChange(key, chart)
```

Comment:
Avoid invoking OnChange directly, enqueue instead.

Reply:
This is not something I added, and I'm concerned that enqueing instead of calling it directly might change the behavior.

Reply:
> This is not something I added, and I'm concerned that enqueing instead of calling it directly might change the behavior.

Ah, sorry. Suppose leaving it would be fine.